client/my-sites/plugins/disconnect-jetpack/disconnect-jetpack-dialog.jsx

```diff
@@ -1,83 +1,95 @@
+import SitesListActions from 'lib/sites-list/actions';
+import { getSelectedSite } from 'state/ui/selectors';
 
-module.exports = React.createClass( {
+class DisconnectJetpackDialog extends Component {
+	constructor( props ) {
+		super( props );
 
-	displayName: 'DisconnectJetpackDialog',
+		this.state = {
+			showJetpackDisconnectDialog: false
+		};
 
-	getInitialState: function() {
-		return { showJetpackDisconnectDialog: false };
-	},
+		this.open = this.open.bind( this );
+		this.close = this.close.bind( this );
+		this.disconnectJetpack = this.disconnectJetpack.bind( this );
+	}
+
+	open() {
+		this.setState( {
+			showJetpackDisconnectDialog: true
+		} );
+	}
 
-	open: function() {
-		this.setState( { showJetpackDisconnectDialog: true } );
-	},
+	close( action ) {
+		this.setState( {
+			showJetpackDisconnectDialog: false
+		} );
 
-	close: function( action ) {
-		this.setState( { showJetpackDisconnectDialog: false } );
 		if ( action === 'continue' ) {
 			this.disconnectJetpack();
 			analytics.ga.recordEvent( 'Jetpack', 'Clicked To Confirm Disconnect Jetpack Dialog' );
 		} else {
 			analytics.ga.recordEvent( 'Jetpack', 'Clicked To Cancel Disconnect Jetpack Dialog' );
 		}
-	},
+	}
 
-	disconnectJetpack: function() {
-		var selectedSite = sites.getSelectedSite();
+	disconnectJetpack() {
+		const { site, selectedSite } = this.props;
 
 		// remove any error and completed notices
 		SitesListActions.removeSitesNotices( [ { status: 'error' }, { status: 'completed' } ] );
 
-		if ( this.props.site ) {
-			SitesListActions.disconnect( this.props.site );
-			if ( selectedSite === this.props.site && this.props.redirect ) {
+		if ( site ) {
+			SitesListActions.disconnect( site );
+			if ( selectedSite === site && this.props.redirect ) {
 				page.redirect( this.props.redirect );
 				return;
 			}
 		} else if ( this.props.sites ) {
-			this.props.sites.getSelectedOrAllWithPlugins().forEach( function( site ) {
-				SitesListActions.disconnect( site );
-			} );
+			this.props.sites.getSelectedOrAllWithPlugins().forEach( siteItem => SitesListActions.disconnect( siteItem ) );
 		}
-		if ( selectedSite === this.props.site ) {
+
+		if ( selectedSite === site ) {
 			page.redirect( '/sites' );
 		}
-	},
+	}
 
-	render: function() {
-		var moreInfo,
-			deactivationButtons = [
-				{
-					action: 'cancel',
-					label: this.translate( 'Cancel' )
-				},
-				{
-					action: 'continue',
-					label: this.translate( 'Disconnect' ),
-					isPrimary: true
-				}
-			];
+	render() {
+		const { site } = this.props;
+		const deactivationButtons = [
+			{
+				action: 'cancel',
+				label: translate( 'Cancel' )
+			},
+			{
+				action: 'continue',
+				label: translate( 'Disconnect' ),
+				isPrimary: true
+			}
+		];
+		let moreInfo;
 
-		if ( this.props.site && this.props.site.name || this.props.site && this.props.site.title ) {
-			moreInfo = this.translate(
+		if ( site && site.name || site && site.title ) {
+			moreInfo = translate(
```

Comment:
Good find. Yeah, it's best if we preserve 2 separate `translate` calls because of the context they provide. But I've moved the logic in a helper method - looks neat now.